.gitignore

```
@@ -0,0 +1,2 @@
+Chart.lock
+charts/
```

```suggestion
charts/*.zip
```
I think having subcharts should be allowed in case we want to refactor that way. But I agree we shouldn't get the zip files (or whichever archive format helm is using)

---

I do not agree. As the subchart's repositories are public we should rely on Chart.lock (which was ignored previously) and dependency versions. Having them inside of our repo is licence sensible, so I would ignore whole charts dir.

---

Well, I think you misunderstand. There are two kinds of subcharts in helm world:

- A distributed `zip` file, e.g. the `minio.zip` you get when enabling it. These should be ignored as they come from the `Chart.lock` file
- Our own "sub-chart", e.g. `api/Chart.yaml`. In my own repo for applications that also depend on external dependencies, I exclude the zips but include my own subcharts.

See: https://git.touhou.fm/touhou.fm/infrastructure/root/-/tree/main/apps/touhou-fm?ref_type=heads

The structure I have there is:

```
- Chart.yaml # the main install
- charts/
  - icecast
  - player
  - playlist
  - provider
  - site
```

But in my dependencies, I mention apollo-graphql router, because this is an integral part of my deployment. I need to combine the APIs from player/playlist/provider to build my full API using the apollo-graphql router. So I deliver (forced) the router, configure it to work with my project, but I **do** ignore the `.zip` because I don't want to have that zip live in git.

---

Also, it's apparently `*.tgz`, see https://git.touhou.fm/touhou.fm/infrastructure/root/-/blob/main/.gitignore?ref_type=heads

---

Yes, I get what you say, I just do not see the actual point for now. The `charts/` is literally empty, so I do not see any reason why we would include anything from charts subdir while it is explicitly meant to be empty. It makes room for mistakes.

---

Well, I mean, directories do not exist in git, as a concept, only files. So you cannot add an empty directory. So as long as it is empty, it will not be created upon `git clone`, nor pushable without contents.

I'm just saying that `chart/*` is too broad and for your mistake prevention should be `chart/*.tgz` instead. This excludes the helm dependencies, does not add or remove the `chart/` subdir. Only when files not matching this, it gets added. So marking it as such, will prevent us mistakingly adding the tgz and getting into license trouble.

Similarly, we might forget to unmark it as ignored and have the same room for mistakes :smile:. This is just based on my knowledge on how git works, and what contents is allowed in `chart/`

---

You do not have to teach how git works, really dude 😄 As you noticed:
> So as long as it is empty

If we mistakenly put something in there, whatever it is, it will end up in our repo. I do not believe that you don't do `git add .` or `git commit -am ...` and then you ensure that all of the added files should be pushed.

In Poland we have a saying
> The turkey was thinking about Sunday, and on Saturday they cut off his head

It sounds stupid in english (it rhymes in Polish), but its meaning is not to think too much forward. I still think, I will repeat myself, that if something is EXPLICITLY meant to be empty, then we HAVE TO ignore the whole thing. Just as a safety precaution in case we put something in there and do `git add .`

> Similarly, we might forget to unmark it as ignored and have the same room for mistakes 😄.

Sure thing, but this will be clearly visible and NOT harmful in case of occurance, as the main branch is protected and contribution has to be made via PRs with at least one reviewer, that is not the author.

---

And I do not think there is any further point that can make me change my mind, so I will resolve this conversation as imo is pointless, we will just try to convince each other to his opinion while all of the shots were fired.

---

In my opinion there is no point in splitting this chart into subcharts now nor in future since most likely we are not going to let people consume only ui of voltaserve.